### DISCUSSION TITLE
Notes fail to appear in Obisidian when page title NULL

### DISCUSSION THREAD

**Discussion by kgmoore431** [coddingtonbear/obsidian-web#181] [Bug Report]
### Describe the bug

It appears that pages that don't have a valid "title" or where the `filename` rendering of "title" is NULL results with a filename of `.md`. This file is created on the local filesystem but is apparently subsequently ignored by Obsidian.  I've found this most often when attempting to bookmark or capture a PDF hosted on a site.
<img width="1895" alt="Screenshot 2024-03-18 at 12 58 15 PM" src="https://github.com/coddingtonbear/obsidian-web/assets/7342515/dee5b807-d525-4c70-a096-244cd97160d3">

In the local filesystem we can see that a "hidden" `.md` file was created and contains our note content:
<img width="854" alt="Screenshot 2024-03-18 at 1 05 03 PM" src="https://github.com/coddingtonbear/obsidian-web/assets/7342515/481186b5-4d37-408c-a62b-bb251e76619a">

Unfortunately this note is nowhere to be found within the Obsidian UI

If I modify the default template API URL to include a suffix to the generated filename e.g. `/vault/Clippings/{{filename page.title}}_clip.md`  I can then see notes appear in Obsidian with the title/filename of `_clip.md` 
<img width="917" alt="Screenshot 2024-03-18 at 1 10 40 PM" src="https://github.com/coddingtonbear/obsidian-web/assets/7342515/82b66c1a-d7e8-4e84-bf10-6c94caf38117">

The downside of this is that it will clobber any existing `_clip.md` note that may have been created previously

### To Reproduce

1. Go to a site that offers a PDF link.  ( e.g. https://www.etcconnect.com/WorkArea/DownloadAsset.aspx?id=10737516212)
2. Activate obsidian web to "Capture Page Screenshot" or "Create new note"
3. Extension reports "All Done! Your note was sent to Obsidiian Successfully!"
4. Obisidian Vault does not show newly created note

### Expected Behavior

Expected to see new note created in Obsidian vault.
I'm not totally sure if this might be more on the REST API plug-in or with the web clipper

### Operating System

OSX Sonoma 14.4 (23E214)

### Browser

Chrome

### Browser Version

122.0.6261.129

**Comment by coddingtonbear** [comment]
I'm afraid this isn't a bug.  You personally have full control over the file name that will be generated and can modify your templates to set whatever title you like -- it's not magic!  I assume in this case, your template is creating a file based on the HTML title (as is the default for the built-in "Capture" template).  All you need to do is modify your template to choose a different algorithm for setting the filename.

**Comment by coddingtonbear** [comment]
I guess I can give one slightly-more-specific recommendation -- if you  look at whatever template you're using, you'll see that it's generating the filename using the page title with something like this:

```handlebars
/vault/Clippings/{{filename page.title}}.md
```

If there is no `page.title`, _of course_ the output of the `filename` helper will be an empty string (because the job of that helper is just to strip out characters that are not safe for a filename).  What you could do, though, is use some simple conditional logic to do the usual thing (above) when the `page.title` field is available, but something else when it isn't.  In the below example, we'll generate the filename as a randomly-generated UUID if no `page.title` exists:

```handlebars
/vault/Clippings/{{#if page.title}}{{filename page.title}}{{else}}{{uuid}}{{/if}}.md
```

There are probably more-useful fallbacks than a UUID, and this is an area where everybody has a different idea about what might be appropriate.  Good news, though: you get to decide instead of me.

**Reply by kgmoore431** [reply]
Thank you for the prompt response.   I do really appreciate the tip for adding a conditional to the filenaming.

**Reply by coddingtonbear** [reply]
No worries & no problem!  Feel free to start another discussion if you run into other questions, too.

**Reply by coddingtonbear** [reply]
Incidentally -- I just noticed I had a typo in my example above, but I've fixed it now!  Specifically -- the `if` is `{{#if page.title}}` -- the `#` is important in the handlebars templating language.
